### ISSUE TITLE
User-written packages fail to import with go.mod. GO111MOD=on

### ISSUE THREAD

**Issue by dreamheld** [Microsoft/vscode-go#1942]
The issue is not with the custom packages vendored through online repos, but with those that you would write for your application yourself. They do not import and are invisible to VSCode.

Steps to Reproduce:

1. Create a new directory (vstest) outside of the current GOPATH. 
2. Navigate to the new directory and run $go mod init vstest
3. Create a subfolder "packtest" with a packtest.go in it.
3.a packtest.go contains:
`package packtest
import "fmt"
//PackPrint tests the autocomplete of user packages in vscode
func PackPrint(){fmt.Println("PackTest Called")`

4. Navigate back up to the vstest folder and create main.go.

5. In main.go try to call packtest.PackPrint(). It fails. Not only VSCode doesn't auto-complete the new package, it doesn't import it either. It also doesn't have it listed in the "Add Import..." option.

(go.autocompleteUnimportedPackages": true)
(env GO111MODULE=on)

Using the import statement like 'import "packtest"' doesn't seem to work either. package main gets highlighted in red with the error "Build vstest: cannot find module for path packtest".

**Comment by lkarlslund** [comment]
This is not a VS Code problem. This is because that's not the way you import packages if you use Go Modules :-)

When using Go, there is no such thing as a relative path. This will not work with "go build main.go" from the commandline either, given your scenario.

When you do an import, Go looks for the package in:

GOROOT/src
GOPATH/src
vendor paths in current folder or parent folders, working its way upwards (if you're inside GOPATH)
modules from go.mod in current folder or parent folders, working its way upwards (if you're outside GOPATH)
.... and nowhere else.

In your go.mod, your own package name is defined in the first line as module <my-module-name>, typically "github.com/username/packagename" if you're using Github at least.

Your import statement in main.go should be:

import "<my-module-name>/packtest"

And you don't need to turn GO111MODULE=on, if you're outside GOPATH it just works from Go 1.11 and onwards.

**Comment by dreamheld** [comment]
> This is not a VS Code problem. This is because that's not the way you import packages if you use Go Modules :-)
...
Right. That's my bad. However, I did try the import "/packtest" statement too this afternoon (just in case it was something along the lines of what you were saying). The terminal would just hang. I'd have to Ctrl+C to break out of that. 

And even w/ the "/packtest" line, there was still no auto-fill of the package methods. :/

**Comment by ramya-rao-a** [comment]
@dreamheld After updating the import statement to "vstest/packtest" and adding a line that uses the package, I am able to get the completions:

![image](https://user-images.githubusercontent.com/16890566/46116120-4d975100-c1af-11e8-84f6-eb94f7291763.png)

**Comment by ramya-rao-a** [comment]
@uudashr In the above code set-up, if I run `Go: Browse Packages`, the `vstest/packtest` doesnt get listed i.e `gopkgs` doesnt return the `vstest/packtest`.

**Comment by dreamheld** [comment]
I wasn't able to respond to this last night, but I think my issue was indeed that I never killed gocode process.

The new beta (as of 20 min ago) seems to do the autocomplete for methods fine, but doesn't fill out the import statement automatically anymore. Also, package/method highlighting for the new package seems off. I'll poke more at it though

**Comment by dreamheld** [comment]
@uudashr how do I reinstall 'gopkgs'? using 'go get'?

**Comment by ramya-rao-a** [comment]
@dreamheld 

> but doesn't fill out the import statement automatically anymore

You mean that in a file where `packtest` is not yet imported, typing `pac` doesnt give you a completion item that would add the import?

If yes, then I can confirm that @uudashr's updated `gopkgs` tool will fix that. 

Run `Go: Install/Update Tools`, select `gopkgs`, press OK. That will update the tool.

**Comment by dreamheld** [comment]
@ramya-rao-a , nope. I've updated @uudashr 's tool and everything works swimmingly with automatically adding the import statement and auto-completing the 'packtest' package name.

The only two issues remaining, it seems:
1. The one you mentioned in a thread above (with method filling showing up only after the method was used already).
2. Highlighting. Method/Package colors seem to not work anymore. Might be an issue on my end though. Still diagnosing that.

**Comment by dreamheld** [comment]
![image](https://user-images.githubusercontent.com/6977284/46172853-ceaf2080-c261-11e8-9e2a-bf8402462129.png)

**Comment by ramya-rao-a** [comment]
I dont see any color issue .....

**Comment by dreamheld** [comment]
Ah, geez! My screen was in the night mode. so everything looked yellow. Duh... I apologize for wasting you time!
